cpp/src/bitmask/null_mask.cu

```diff
@@ -222,6 +282,42 @@ cudf::size_type count_unset_bits(bitmask_type const *bitmask, size_type start,
   return (num_bits - detail::count_set_bits(bitmask, start, stop, stream));
 }
 
+// Create a bitmask from a vector of column views
+void concatenate_masks(std::vector<column_view> const &views,
+    bitmask_type * dest_mask,
+    cudaStream_t stream) {
+  using CDViewPtr =
+    std::unique_ptr<column_device_view,
+                    std::function<void(column_device_view*)>>;
+  std::vector<CDViewPtr> cols;
+  thrust::host_vector<column_device_view> deviceViews;
+
+  thrust::host_vector<size_type> view_offsets(1, 0);
+  for (auto &v : views) {
+    cols.emplace_back(column_device_view::create(v, stream));
+    deviceViews.push_back(*(cols.back()));
+    view_offsets.push_back(v.size());
+  }
+  thrust::inclusive_scan(thrust::host,
+      view_offsets.begin(), view_offsets.end(),
+      view_offsets.begin());
+
+  rmm::device_vector<column_device_view> dViews = deviceViews;
+  rmm::device_vector<size_type> dOffsets = view_offsets;
+
+  column_device_view* dViewsPtr = thrust::raw_pointer_cast(dViews.data());
+  size_type* dOffsetPtr = thrust::raw_pointer_cast(dOffsets.data());
+
+  auto number_of_mask_bits = view_offsets.back();
+  cudf::util::cuda::grid_config_1d config(number_of_mask_bits, 256);
+  concatenate_masks_kernel<<<config.num_blocks, config.num_threads_per_block,
```

Comment:
```suggestion
  concatenate_masks_kernel<<<config.num_blocks, block_size,
```

Reply:
Not to be rude but why is this better?

Reply:
Many of our kernels take the block_size as a template parameter as an optimization, rather than using `blockDim.x` in the kernel. But if you don't want to do that, it's fine.

Reply:
No that sounds fair. I was under the impression that this was a silly mistake.